sdk/core/azure-core/CMakeLists.txt

```diff
@@ -47,7 +47,9 @@ set(
     ${WIN_TRANSPORT_ADAPTER_INC}
     inc/azure/core/credentials/credentials.hpp
     inc/azure/core/credentials/token_credential_options.hpp
+    inc/azure/core/cryptography/hash_base.hpp
     inc/azure/core/cryptography/hash.hpp
+    inc/azure/core/cryptography/md5.hpp
```

Comment:
Add explicitly to the simplified header and test.

Reply:
That said, I recommend reverting this change and keeping everything in the single `hash.hpp` header file. People on the C++ team had advised **against** too granular header files (for example one per type), and were in favor of fewer, bigger header files (unless you expect that header would end up so big that compiler throughput would be measurably damaged by doing so). This is also easier for the end user and reduces a bunch of indirection and links within the generated docs.

Reply:
More importantly, it is difficult for me to see and understand what exactly is changing because of this file split, in this PR. It makes reviewing code and logic difficult (I end up reviewing other things like doc comments, etc.). If you feel strongly about splitting it, please do it in a separate PR.

Reply:
> keeping everything in the single hash.hpp

As we keep adding Hashing algorithms, we should be able to include all of them or just the one we want.
We will need to add SHA next and maybe others. Let's make it easy for developers and also for maintainers :)

Reply:
> More importantly, it is difficult for me to see and understand what exactly is changing

I can show/tell the main changes and explain any question if you want.  I wouldn't really like to revert XD,  I would prefer to let it up to whatever you'd like to do :)

Reply:
Actually, there is only one change to the header for MD5 class.. other than that, it's the same.

Reply:
So, it might not be that hard to let just one header.